src/api.ts

```diff
@@ -1,68 +1,56 @@
 import * as u from './util'
-var EventEmitter = require('events')
-var Hookable = require('hoox')
+const EventEmitter = require('events')
+const Hookable = require('hoox')
 
-function id<T = any> (e: T): T {
-  return e
-}
+const identity = (x: any): any => x
 
 function merge (a: any, b: any, mapper?: any) {
-  mapper = mapper || id
-
-  for (var k in b) {
+  mapper = mapper ?? identity
```

Comment:
yeah just tested it, this breaks in node 10 + 12, which are both LTS versions I would expect to work.

I don't know what the output of this .ts code is though ...

Reply:
@mixmix No worries, TypeScript compiles down to ES5, so we're looking good in terms of support. I considered configuring TypeScript to compile to ES6, which would support node 10 and above, and *partially support* node 5, but I think it's important to keep the support broad when it comes to secret-stack.

Here's how `src/api.ts` compiles down to `lib/api.js` (first lines where it's relevant to your comment):

```js
"use strict";
var u = require("./util");
var EventEmitter = require('events');
var Hookable = require('hoox');
function id(e) {
    return e;
}
function merge(a, b, mapper) {
    mapper = mapper || id;
    for (var k in b) {
        if (b[k] &&
            typeof b[k] === 'object' &&
            !Buffer.isBuffer(b[k]) &&
            !Array.isArray(b[k])) {
            merge((a[k] = {}), b[k], mapper);
        }
        else {
            a[k] = mapper(b[k], k);
        }
    }
    return a;
}

...
```

Reply:
Oh that's awesome news